azdev/operations/linter/linter.py

```diff
@@ -223,22 +257,65 @@ def run(self, run_params=None, run_commands=None, run_command_groups=None, run_h
 
         if not self.exit_code:
             print(os.linesep + 'No violations found.')
+
+        if self._update_global_exclusion is not None:
+            repo_paths = None
+            if self._update_global_exclusion == 'main':
+                repo_paths = [get_cli_repo_path()]
+            else:
+                repo_paths = get_ext_repo_paths()
+            exclusion_paths = [os.path.join(repo_path, 'linter_exclusions.yml') for repo_path in repo_paths]
```

Comment:
edge case: If user don't `azdev setup` with `-r`, the `repo_paths` would be `None`, then it would be not iterable, such that this azdev would crash in this line.